### PR TITLE
Simple backend chapters

### PR DESCRIPTION
**Pull request by mark-i-m**
This adds the basics noted by @nagisa here: https://gist.github.com/nagisa/a311a0dab09851397f266076130eefb6. Some of it is copied nearly verbatim from their notes. (Thanks @nagisa !)

We are looking into lectures/meetings/discussions to try to flesh this out a bit more, but for now, it would be good to have this content, I think. Feedback welcome!

### CODE REVIEW

File: src/backend/monomorph.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+fn banana() {
+   peach::<u64>();
+}
+
+fn main() {
+    banana();
+}
+```
+
+The monomorphisation collector will give you a list of `[main, banana,
```
Comment (JohnTitor) [inline src/backend/monomorph.md]:
```suggestion
The monomorphization collector will give you a list of `[main, banana,
```

File: src/backend/monomorph.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+peach::<u64>]`. These are the functions that will have machine code generated
+for them. Collector will also add things like statics to that list.
+
+See [the collector rustdocs][collect] for more info.
+
+[collect]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir/monomorphize/collector/index.html
+
+The monomorphisation collector is run just before MIR lowering and codegen.
```
Comment (JohnTitor) [inline src/backend/monomorph.md]:
```suggestion
The monomorphization collector is run just before MIR lowering and codegen.
```

File: src/backend/monomorph.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+[`rustc_codegen_ssa::base::codegen_crate`][codegen1] calls the
+[`collect_and_partition_mono_items`][mono] query, which does monomorphisation
```
Comment (JohnTitor) [inline src/backend/monomorph.md]:
```suggestion
[`collect_and_partition_mono_items`][mono] query, which does monomorphization
```

File: src/backend/monomorph.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+collection and then partitions them into [codegen
+units](../appendix/glossary.md).
+
+[mono]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir/monomorphize/partitioning/fn.collect_and_partition_mono_items.html
+[codegen1]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_codegen_ssa/base/fn.codegen_crate.html
+
+## Polymorphization
+
+As mentioned above, monomorphisation produces fast code, but it comes at the
```
Comment (JohnTitor) [inline src/backend/monomorph.md]:
```suggestion
As mentioned above, monomorphization produces fast code, but it comes at the
```

File: src/backend/lowering-mir.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+`assert` can result in multiple basic blocks. This is a perfect lede into the
+non-portable LLVM-specific part of the code generation. Intrinsic generation is
+fairly easy to understand as it involves very few abstraction levels in between
+and can be found in [`rustc_codegen_llvm::intrinsic`][llvmint].
+
+[llvmint]: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_codegen_llvm/intrinsic/index.html
+
+Everything else will use the [builder interface][builder], this is the code that gets
+called in [`librustc_codegen_ssa::mir::*`][ssamir] modules that was discussed
+above.
```
Comment (mark-i-m) [inline src/backend/lowering-mir.md]:
```suggestion
Everything else will use the [builder interface][builder]. This is the code that gets
called in the [`librustc_codegen_ssa::mir::*`][ssamir] modules discussed above.
```

Comment (mark-i-m) [reply src/backend/lowering-mir.md]: Also, lol that this is the easiest way to make a minor edit to my own PR...

**Pull request merged**